DashAI/back/api/api_v1/endpoints/global_explanation.py

```
@@ -0,0 +1,198 @@
+import logging
+from typing import Union
+
+from fastapi import APIRouter, Depends, status
+from fastapi.exceptions import HTTPException
+from sqlalchemy import exc, select
+from sqlalchemy.orm import Session
+
+from DashAI.back.api.api_v1.schemas.explanation_params import GlobalExplanationParams
+from DashAI.back.api.deps import get_db
+from DashAI.back.database.models import GlobalExplanation, Run
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.get("/")
+async def get_global_explanations(
+    run_id: Union[int, None] = None,
+    db: Session = Depends(get_db),
+):
+    """Return the available global explanations in the database.
+    The global explanations can be filtered by run_id.
+
+    Parameters
+    ----------
+    run_id: Union[int, None], optional
+        If specified, the function will return all the explainers associated with
+        the run, by default None.
+    Returns
+    -------
+    List[dict]
+        A list of dict containing explainers.
+
+    Raises
+    ------
+    HTTPException
+    """
+    try:
+        if run_id is not None:
+            global_explanations = db.scalars(
+                select(GlobalExplanation).where(GlobalExplanation.run_id == run_id)
+            ).all()
+        else:
+            global_explanations = db.query(GlobalExplanation).all()
```

Will it ever be necessary to bring absolutely all explanations?

I think that for security, I would just throw an exception when run_id is not there, instead of delivering all possible explanations.